src/Chapter3.hs

```diff
@@ -480,6 +527,28 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = Castle String
+data ChurchOrLibrary = Church | Library
+data HouseMembers = One | Two | Three | Four
+
+data City = City
+    { cityCastle :: Castle
+    , cityWall :: Bool
+    , churchOrLib :: ChurchOrLibrary
+    , homes :: [HouseMembers]
+    }
+
+buildCastle :: String -> City -> City
+buildCastle castle city = city {cityCastle = Castle castle}
+
+buildHouse :: HouseMembers -> City -> City
+buildHouse hm city = city {homes = hm : homes city}
+
+buildWalls :: City -> City
+buildWalls city = case cityCastle city of
+    Castle _ -> city {cityWall = True}
```

Comment:
In this function you don't check whether the city has enough people :thinking: It looks like you should count the total number of people and compare it here :slightly_smiling_face: